code/modules/assembly/health.dm

```diff
@@ -30,56 +30,42 @@
 
 /obj/item/device/assembly/health/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if(istype(W, /obj/item/device/multitool))
-		if(alarm_health == 0)
-			alarm_health = -90
-			user.show_message("You toggle [src] to \"detect death\" mode.")
-		else
-			alarm_health = 0
-			user.show_message("You toggle [src] to \"detect critical state\" mode.")
+		alarm_health = Clamp(input(user,"Input health variable to detect from -90 to 80 (0 is crit state, 80 is hurt, -90 is death)",src,0) as num, -90, 80)
 		return
-	else
-		return ..()
+	..()
 
-/obj/item/device/assembly/health/process()
-	if(!scanning || !secured)
+/obj/item/device/assembly/health/HasProximity(mob/living/M)
+	if(!secured || !scanning || cooldown > 0)
 		return
-
-	var/atom/A = src
-	if(connected && connected.holder)
-		A = connected.holder
-
-	for(A, A && !ismob(A), A=A.loc);
-	// like get_turf(), but for mobs.
-	var/mob/living/M = A
-
-	if(M)
-		health_scan = M.health
-		if(health_scan <= alarm_health)
-			pulse()
-			audible_message("\icon[src] *beep* *beep*", "*beep* *beep*")
-			toggle_scan()
+	if(!istype(M))
 		return
-	return
+	health_scan = M.health
+	if(health_scan <= alarm_health)
+		pulse()
+		audible_message("\icon[src] *beep* *beep*", "*beep* *beep*")
+		cooldown = 2
+		spawn(10)
+			process_cooldown()
 
 /obj/item/device/assembly/health/proc/toggle_scan()
 	if(!secured)	return 0
 	scanning = !scanning
 	if(scanning)
-		SSobj.processing |= src
+		SSobj.processing.Add(src)
```

Comment:
|= is way safer than Add, or the object could go inside processing list twice if shit happens